pkg/spoke/addon/lease_controller.go

```diff
@@ -0,0 +1,191 @@
+		if err != nil {
+			return err
+		}
+		errs := []error{}
+		for _, addOn := range addOns {
+			addOnConfig, err := getAddOnConfig(addOn)
+			if err != nil {
+				// failed to get addon configuration, ignore it.
+				continue
+			}
+			if err := c.syncSingle(ctx, addOnConfig.InstallationNamespace, addOn, syncCtx.Recorder()); err != nil {
+				errs = append(errs, err)
+			}
+		}
+		return operatorhelpers.NewMultiLineAggregate(errs)
+	}
+
+	addOnNamespace, addOnName, err := cache.SplitMetaNamespaceKey(queueKey)
+	if err != nil {
+		// queue key is bad format, ignore it.
+		return nil
+	}
+
+	addOn, err := c.addOnLister.ManagedClusterAddOns(c.clusterName).Get(addOnName)
+	if errors.IsNotFound(err) {
+		// addon is not found, could be deleted, ignore it.
+		return nil
+	}
+	if err != nil {
+		return err
+	}
+
+	return c.syncSingle(ctx, addOnNamespace, addOn, syncCtx.Recorder())
+}
+
+func (c *managedClusterAddOnLeaseController) syncSingle(ctx context.Context,
+	leaseNamespace string,
+	addOn *addonv1alpha1.ManagedClusterAddOn,
+	recorder events.Recorder) error {
+	// addon lease name should be same with the addon name.
+	observedLease, err := c.leaseLister.Leases(leaseNamespace).Get(addOn.Name)
+
+	var condition metav1.Condition
+	switch {
+	case errors.IsNotFound(err):
+		condition = metav1.Condition{
+			Type:    addOnAvailableConditionType,
+			Status:  metav1.ConditionUnknown,
+			Reason:  "ManagedClusterAddOnLeaseNotFound",
+			Message: "Managed cluster addon agent lease is not found.",
+		}
+	case err != nil:
+		return err
+	case err == nil:
+		gracePeriod := time.Duration(leaseDurationTimes*leaseDurationSeconds) * time.Second
+		if time.Now().Before(observedLease.Spec.RenewTime.Add(gracePeriod)) {
+			// the lease is constantly updated, update its addon status to available
+			condition = metav1.Condition{
+				Type:    addOnAvailableConditionType,
+				Status:  metav1.ConditionTrue,
+				Reason:  "ManagedClusterAddOnLeaseUpdated",
+				Message: "Managed cluster addon agent updates its lease constantly.",
+			}
+			break
+		}
+
+		// the lease is not constantly updated, update its addon status to unavailable
+		condition = metav1.Condition{
+			Type:    addOnAvailableConditionType,
+			Status:  metav1.ConditionFalse,
+			Reason:  "ManagedClusterAddOnLeaseUpdateStopped",
+			Message: "Managed cluster addon agent stopped updating its lease.",
+		}
+	}
+
+	if meta.IsStatusConditionPresentAndEqual(addOn.Status.Conditions, condition.Type, condition.Status) {
+		// addon status is not changed, do nothing
+		return nil
+	}
+
+	_, updated, err := helpers.UpdateManagedClusterAddOnStatus(
+		ctx,
+		c.addOnClient,
+		c.clusterName,
+		addOn.Name,
+		helpers.UpdateManagedClusterAddOnStatusFn(condition),
+	)
+	if err != nil {
+		return err
+	}
+	if updated {
+		recorder.Eventf("ManagedClusterAddOnStatusUpdated",
+			"update managed cluster addon %q available condition to unavailable, due to its lease is not updated constantly",
```

Comment:
I think you can update it to true and unknown as well.

Reply:
yeah, good catch